File: src/main/java/io/github/malczuuu/iemu/domain/CurrentTimeModule.java
package io.github.malczuuu.iemu.domain;

import java.time.Instant;
import java.util.function.Consumer;

public interface CurrentTimeModule {

  void setCurrentTime(Instant currentTime);

  void setTimeZone(String timeZone);

  void setUTCOffset(String utcOffset);

  Instant getCurrentTime();

  String getTimeZone();

  String getUTCOffset();

  void shutdown();

  void subscribeOnCurrentTimeChange(Consumer<Instant> consumer);
}

File: src/main/kotlin/io/github/malczuuu/iemu/domain/CurrentTimeModule.kt
package io.github.malczuuu.iemu.domain

import java.time.Instant

interface CurrentTimeModule {

  var timeZone: String

  var utcOffset: String

  var currentTime: Instant

  fun subscribeOnCurrentTimeChange(consumer: (Instant) -> Unit): Boolean

  fun shutdown()
}
